src/uwtools/config/jinja2.py

```diff
@@ -310,7 +310,7 @@ def path_join(path_components: List[str]) -> str:
             raise UndefinedError()
         return os.path.join(*path_components)
 
-    filters = dict(path_join=path_join)
+    filters = dict(env=lambda var: os.environ[var], path_join=path_join)
```

Comment:
This adds an `env` filter to access specific environment variables in configs. For example, in `file copy` or `file link` mode, we could now have e.g.
```
files:
  f000.00: "{{ basedir }}/{{ 'YYYYMMDD' | env }}/00/gfs.t00z.pgrb2.0p25.f000"
  f012.00: "{{ basedir }}/{{ 'YYYYMMDD' | env }}/12/gfs.t12z.pgrb2.0p25.f000"
basedir: /path/to/data
```
in `config.yaml`. The hardcoded `basedir` value keeps that path DRY, and the reference to the `YYYYMMDD` environment variable would support an invocation like
```
YYYYMMDD=20240509 uw file link --config-file config.yaml --target-dir target files
```
While working on the video demo for the `file` tool, this seemed like something you'd obviously want to do when using that, to parameterize a file copy/link list by e.g. cycle. I'm reluctant to litter our use of Jinja2 with custom filters and functions (we currently only have one other filer: `path_join` filter, to create filesystem paths from components) but this one seems maybe indispensable. But if anyone has an idea how else we might do this -- or thinks that we shouldn't -- please let me know.

BTW, it would also be possible to add custom _functions_ to Jinja2, instead of filters. For example, instead of `{{ 'YYYYMMDD' | env }}`, we could have `{{ env('YYYYMMDD') }}`. I can't immediately find a reason to prefer one over the other, except that the filter version is more Jinja2-ish.

Reply:
I like the addition of the `env` filter. I think many of our users will appreciate it. 

I'm curious if we might also entertain adding an optional `--cycle` flag to this tool so that folks who want to pull pieces directly from driver config files would have a similar experience with this tool. It might be nice to have an option that is configurable in the template.

Reply:
That seems like a good idea, but probably a bit of work and so then probably worthy of a ticket and scheduling. We might as well add `--leadtime` then, too, to support all driver config YAML. I'll add a Jira ticket.